packages/authenticator/amplify_authenticator/lib/src/services/amplify_auth_service.dart

```diff
@@ -311,13 +313,24 @@ class AmplifyAuthService implements AuthService {
   }
 
   @override
-  Future<AmplifyConfig> waitForConfiguration() {
-    return Amplify.asyncConfig;
+  Future<AmplifyConfig> waitForConfiguration() async {
+    final timer = Timer.periodic(const Duration(seconds: 10), (timer) {
+      logger.warn(
+        'Amplify is taking longer to configure than expected.'
+        ' Have you called `Amplify.configure()`?',
+      );
+    });
+    final config = await Amplify.asyncConfig;
+    timer.cancel();
+    return config;
```

Comment:
`asyncConfig` can throw so we probably want to do:

```suggestion
    try {
      return await Amplify.asyncConfig;
    } finally {
      timer.cancel();
    }
```